src/progress.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef SRC_CPP_PROGRESS_HPP_
+#define SRC_CPP_PROGRESS_HPP_
+
+#include <iostream>
+
+void progress(int phase, int n, int max_n) {
+    float p = (100.0 / 4) * ((phase - 1.0) + (1.0 * n / max_n));
+    std::cout << "Progress: " << p << std::endl;
```

Comment:
I'm a fan of:

```
std::cout << "\rProgress: " << p << "  ";
std::cout.flush();
```

In case this is printed frequently, that won't make your terminal scroll so much

Reply:
you may also want `<< std::setprecision(2) << (p * 100.0) << "% "`

Reply:
That's a good idea. I'll make that change.

Reply:
I think we still need a line break after p, so that you don't see this:

```
Progress: 42.0F1 complete, time: 0.011 seconds. CPU (151.67%) Mon Mar 22 23:18:05 2021
```

on the same line